src/Interfacer.jl

```diff
@@ -242,4 +248,24 @@ reinit!(sim::ComponentModelSimulation) = error("undefined reinit! for " * name(s
 # Include file containing the surface stub simulation type.
 include("surface_stub.jl")
 
+"""
+    AbstractModeType
+
+An abstract type representing a simulation mode.
+"""
+abstract type AbstractSimulationMode end
+
+"""
+    AbstractSlabplanetSimulationMode
+
+An abstract type representing a simulation mode for slabplanet models.
+"""
+abstract type AbstractSlabplanetSimulationMode <: AbstractSimulationMode end
+
+struct AMIPMode <: AbstractSimulationMode end
+struct SlabplanetMode <: AbstractSlabplanetSimulationMode end
+struct SlabplanetAquaMode <: AbstractSlabplanetSimulationMode end
+struct SlabplanetTerraMode <: AbstractSlabplanetSimulationMode end
+struct SlabplanetEisenmanMode <: AbstractSlabplanetSimulationMode end
```

Comment:
It would be good to include short comments about each of these simulation setups. You can get this information from experiments/ClimaEarth/README.md